parser/src/java/org/apache/hadoop/hive/ql/parse/IdentifiersParser.g

```diff
@@ -734,6 +734,21 @@ dropPartitionOperator
     EQUAL | NOTEQUAL | LESSTHANOREQUALTO | LESSTHAN | GREATERTHANOREQUALTO | GREATERTHAN
     ;
 
+filterPartitionSpec
+    :
+    LPAREN filterPartitionVal (COMMA  filterPartitionVal )* RPAREN -> ^(TOK_PARTSPEC filterPartitionVal +)
+    ;
+
+filterPartitionVal
+    :
+    identifier filterPartitionOperator constant -> ^(TOK_PARTVAL identifier filterPartitionOperator constant)
```

Comment:
old `partitionSpec` doesn't mandatorily required the constant
```
identifier (EQUAL constant)? 
```

were there any use cases of that?

Reply:
Based on the implementation, I don't think there was never such a use case. I guess we always except it to be contatant
Refer:
https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/ddl/misc/msck/MsckAnalyzer.java#L66
https://github.com/apache/hive/blob/6f9c7bc490e31ee23556a316f2ac4a886d4eff1e/ql/src/java/org/apache/hadoop/hive/ql/parse/BaseSemanticAnalyzer.java#L1527
https://github.com/apache/hive/blob/6f9c7bc490e31ee23556a316f2ac4a886d4eff1e/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreChecker.java#L259